knimin/lib/tests/test_data_access.py

```diff
@@ -1260,6 +1301,142 @@ def test_extract_sample_plates(self):
                'notes': None}
         self.assertEqual(obs_info, exp)
 
+    def test_normalize_shotgun_plate_bad_id(self):
+        with self.assertRaisesRegexp(ValueError, "shotgun plate"):
+            db.normalize_shotgun_plate(99999999, 'test', 'a valid echo name',
+                                       np.zeros((16, 24)), np.zeros((16, 24)))
+
+    def test_normalize_shotgun_plate_bad_water_shape(self):
+        # this test assumes that the plate is a 384 well format, so we'll specify
+        # a water matrix in 96 well format
+        self._create_test_echo()
+        cid = self._create_test_shotgun_plate()
+        with self.assertRaisesRegexp(ValueError, "plate_normalization_water"):
+            db.normalize_shotgun_plate(cid, 'test', 'a valid echo name',
+                                       np.zeros((16, 24)), np.zeros((8, 12)))
+
+    def test_normalize_shotgun_plate_bad_sample_shape(self):
+        # this test assumes that the plate is a 384 well format, so we'll specify
+        # a sample matrix in 96 well format
+        self._create_test_echo()
+        cid = self._create_test_shotgun_plate()
+        with self.assertRaisesRegexp(ValueError, "plate_normalization_sample"):
+            db.normalize_shotgun_plate(cid, 'test', 'a valid echo name',
+                                       np.zeros((8, 12)), np.zeros((16, 24)))
+
+    def test_normalize_shotgun_plate_bad_echo_name(self):
+        ### there does not appear to be any loaded echo names in patch 0041
+        cid = self._create_test_shotgun_plate()
+        with self.assertRaisesRegexp(ValueError, "echo machine"):
+            db.normalize_shotgun_plate(cid, 'test', 'does not exist',
+                                       np.zeros((16, 24)), np.zeros((16, 24)))
+
+    def test_normalize_shotgun_plate(self):
+        before = datetime.datetime.now()
+        self._create_test_echo()
+        cid = self._create_test_shotgun_plate()
+        nid = db.normalize_shotgun_plate(cid, 'test', 'a valid echo name',
+                                         np.arange(384).reshape(16, 24),
+                                         np.arange(384).reshape(16, 24) * 10)
+        after = datetime.datetime.now()
+        exp_sample = np.arange(384).reshape(16, 24)
+        exp_water = np.arange(384).reshape(16, 24) * 10
+
+        ### many of these values are ignored at this time as dont know
```

Comment:
You can set them to None, and test the entire `dict`. That is the expected behavior since that information is loaded at 2 different steps of the pipeline.